File: src/services/customerService.ts
import api from './api';
import type { Customer, PaginatedResponse, PaginationParams } from '@/types';

export const customerService = {
  async getAll(params?: Partial<PaginationParams>): Promise<PaginatedResponse<Customer>> {
    const { data } = await api.get('/customers', { params });
    return data;
  },

  async getById(id: string): Promise<Customer> {
    const { data } = await api.get(`/customers/${id}`);
    return data;
  },

  async create(customer: Partial<Customer>): Promise<Customer> {
    const { data } = await api.post('/customers', customer);
    return data;
  },

  async update(id: string, customer: Partial<Customer>): Promise<Customer> {
    const { data } = await api.put(`/customers/${id}`, customer);
    return data;
  },

  async delete(id: string): Promise<void> {
    await api.delete(`/customers/${id}`);
  },
};

File: FieldServiceManangement-main/src/main/java/com/fieldservicemanagement/field_service_management/service/CustomerService.java
package com.fieldservicemanagement.field_service_management.service;

import com.fieldservicemanagement.field_service_management.entity.Customer;
import com.fieldservicemanagement.field_service_management.repository.CustomerRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.NoSuchElementException;

@Service
public class CustomerService {

    private final CustomerRepository customerRepository;

    public CustomerService(CustomerRepository customerRepository) {
        this.customerRepository = customerRepository;
    }

    public List<Customer> getAll() {
        return customerRepository.findAll();
    }

    public Customer getById(Long id) {
        return customerRepository.findById(id)
                .orElseThrow(() -> new NoSuchElementException("Customer not found with id: " + id));
    }

    public Customer create(Customer customer) {
        return customerRepository.save(customer);
    }

    public Customer update(Long id, Customer customerDetails) {
        Customer customer = getById(id);
        customer.setCompanyName(customerDetails.getCompanyName());
        customer.setContactName(customerDetails.getContactName());
        customer.setEmail(customerDetails.getEmail());
        customer.setPhone(customerDetails.getPhone());
        customer.setAddress(customerDetails.getAddress());
        return customerRepository.save(customer);
    }

    public void delete(Long id) {
        customerRepository.deleteById(id);
    }
}